django/utils/html.py

```diff
@@ -250,25 +250,30 @@ def __call__(self, text, trim_url_limit=None, nofollow=False, autoescape=False):
 
         If autoescape is True, autoescape the link text and URLs.
         """
-        self.trim_url_limit = trim_url_limit
-        self.nofollow = nofollow
-        self.autoescape = autoescape
-        self.safe_input = isinstance(text, SafeData)
+        safe_input = isinstance(text, SafeData)
 
         words = self.word_split_re.split(str(text))
         return ''.join([
-            self.handle_word(word) for word in words
+            self.handle_word(
+                word,
+                safe_input=safe_input,
+                trim_url_limit=trim_url_limit,
+                nofollow=nofollow,
+                autoescape=autoescape,
+            ) for word in words
         ])
 
-    def handle_word(self, word):
+    def handle_word(
+        self, word, safe_input, trim_url_limit=None, nofollow=False, autoescape=False,
+    ):
```

Comment:
Ensure that the new arguments are keyword only:

```suggestion
    def handle_word(
        self, word, *, safe_input, trim_url_limit=None, nofollow=False, autoescape=False,
    ):
```